### ISSUE TITLE
 PHP FPM client limit under debian apache is by default fairly limited, not using available slots 

### ISSUE THREAD

**Issue by hayden-t** [virtualmin/virtualmin-install#113]
This needs fixing, i posted ages ago in gpl (with no response) but realise not it might belong here ?

https://github.com/virtualmin/virtualmin-gpl/issues/680

**Comment by jcameron** [comment]
Which Linux distro and version are you seeing this on?

**Comment by hayden-t** [comment]
debian 11 but it would be older ones too, TasksMax defaults to 614 whih works out to be about apache 300 slots which is not much.

in munin it looked like this, where i had plenty of free slots (600) but server was very slow to respond as only 300 being allocated and all used. my stack post linked has more details

**Comment by jcameron** [comment]
Is that task limit of 614 the default for a new debian 11 install? It seems really low to me ..

**Comment by hayden-t** [comment]
yes default for several version

**Comment by iliajie** [comment]
I don’t think this is something Virtualmin should handle! We rely on system defaults, leaving any necessary manual configuration to the user.

**Comment by hayden-t** [comment]
not really, there are many system settings that are set-table via vmin/webmin, there needs to be at least a note about it in the mpm prefork config file, as it was very difficult and obscure for me to work this out.

**Comment by iliajie** [comment]
I still don’t think we should touch it! Besides, on my systems, including Ubuntu 24.04 and Debian 11, the default `TasksMax` is already set to `1610`.

**Comment by hayden-t** [comment]
Im running debian 11 and its not, it was 614

**Comment by iliajie** [comment]
> Im running debian 11 and its not, it was 614

The default Debian 11 installation sets `TasksMax` to `1610`, e.g.:

<img width="550" alt="image" src="https://github.com/user-attachments/assets/43642da1-e6d7-4902-8174-360e84b7de8e">

**Comment by hayden-t** [comment]
ill check this out later, about to go to work

**Comment by hayden-t** [comment]
```
root@vps:~# systemctl show apache2.service | grep TasksMax
TasksMax=2000
root@vps:~# systemctl show mysql | grep TasksMax
mysql.service   mysqld.service
root@vps:~# systemctl show mysqld.service | grep TasksMax
TasksMax=614
root@vps:~# cat /etc/deb
debconf.conf    debian_version
root@vps:~# cat /etc/debian_version
11.10
root@vps:~# systemctl show bind9.service | grep TasksMax
TasksMax=614
root@vps:~#

```
as you can see the default for me is 614

googling i found this 
https://unix.stackexchange.com/a/606524

```
     The system default for this setting may be controlled with DefaultTasksMax= 
      in systemd-system.conf(5).
```
and then this 
https://manpages.debian.org/testing/systemd/systemd-system.conf.5.en.html

`DefaultTasksMax=
Configure the default value for the per-unit TasksMax= setting. See [systemd.resource-control(5)](https://manpages.debian.org/testing/systemd/systemd.resource-control.5.en.html) for details. This setting applies to all unit types that support resource control settings, with the exception of slice units. Defaults to 15% of the minimum of kernel.pid_max=, kernel.threads-max= and root cgroup pids.max. Kernel has a default value for kernel.pid_max= and an algorithm of counting in case of more than 32 cores. For example, with the default kernel.pid_max=, DefaultTasksMax= defaults to 4915, but might be greater in other systems or smaller in OS containers.
`

so it seems its somewhat dynamic depending on the system which for me :
root@vps:~# cat /proc/sys/kernel/pid_max
4194304

so not sure how it becomes 614

**Comment by hayden-t** [comment]
for my own records and of interest i found the possible 'origin' of mpm_prefork.conf 
https://salsa.debian.org/apache-team/apache2/-/blob/master/debian/config-dir/mods-available/mpm_prefork.conf?ref_type=heads

**Comment by iliajie** [comment]
> The system default for this setting may be controlled with DefaultTasksMax= 
>       in systemd-system.conf(5).

I wouldn’t overcomplicate it and just tweaked that in `/etc/systemd/system.conf` file! To clarify, by default, `DefaultTasksMax` is set to `15%` on Debian and its derivatives, while it’s 80% on EL derivatives like Alma and Rocky.

You could technically control it globally, but I’d recommend being cautious.

Either way, this isn’t a Virtualmin issue. And, thank you for the heads up—I’m closing this ticket now.

**Comment by hayden-t** [comment]
I dont need to change the system wide default, its just for apache, Id just thought it could be beneficial for your project to find someway to alert users to the possibility that their web server is being throttle by this arbitrary limit.

**Comment by iliajie** [comment]
> I dont need to change the system wide default

Fair point!

> Id just thought it could be beneficial for your project to find someway to alert users to the possibility that their web server is being throttle by this arbitrary limit.

Well, no one else has ever brought this up. To be clear, the defaults are there for a reason. I don’t think we should micromanage this. If we start handling this specific case, it raises the question of why we wouldn’t do the same for many others—it's a rabbit hole.

**Comment by hayden-t** [comment]
it might be more a webmin thing ?

**Comment by iliajie** [comment]
Well, yes, if we had a dedicated systemd module.
